packages/excalidraw/components/App.tsx

```diff
@@ -4265,7 +4263,37 @@ class App extends React.Component<AppProps, AppState> {
         !this.state.selectionElement &&
         !this.state.selectedElementsAreBeingDragged
       ) {
-        const shape = findShapeByKey(event.key);
+        let shape = findShapeByKey(event.key);
+        //if shapes are selected and waht to change ao of them togather
+        const selectedShape = this.scene.getSelectedElements({
+          selectedElementIds: this.state.selectedElementIds,
+          includeBoundTextElement: true,
+          includeElementsInFrames: true,
+        });
+        if (selectedShape.length > 0 && shape) {
+          const newShapeArr = [];
+          for (let i = 0; i < selectedShape.length; i++) {
+            const found_shape = selectedShape[i];
+            if (
+              found_shape.type !== shape &&
+              (shape === "ellipse" ||
+                shape === "diamond" ||
+                shape === "rectangle") &&
+              ["ellipse", "diamond", "rectangle"].includes(found_shape.type)
+            ) {
+              const newShape = newElement({ ...found_shape, type: shape });
```

Comment:
`mutateElement()` allows changing the type attribute and if it works, it will take care of the history undo/redo as well. You won't even need to update the scene because it takes care of that too.